src/lib/saml-authentication.ts

```diff
@@ -0,0 +1,52 @@
+import * as saml from 'samlify';
+import { IdentityProvider } from 'samlify/types/src/entity-idp';
+import { ServiceProvider } from 'samlify/types/src/entity-sp';
+import { ServiceProviderSettings, IdentityProviderSettings } from 'samlify/types/src/types';
+
+// service proivder
+// Our node applications is the service provider
+export function createServiceProvider(xmlMetaData: string | Buffer, args: ServiceProviderSettings = {}): ServiceProvider {
+  return saml.ServiceProvider({ ...args, metadata: xmlMetaData });
+}
+
+// The external party
+export function createIdentityProvider(xmlMetaData: string | Buffer, args: IdentityProviderSettings = {}): IdentityProvider {
+  return saml.IdentityProvider({ ...args, metadata: xmlMetaData });
+}
+
+// Flow
+
+// 1. The user tries to log in to service provider from a browser
+
+// 2. Service provider generates SAML REQUEST
+export function createLoginRequest(sp: ServiceProvider, idp: IdentityProvider, customTagReplacement: string = 'redirect') {
+  console.log('fsd', idp);
+  const { context } = sp.createLoginRequest(idp, customTagReplacement);
+  return context;
+}
+
+// 3. The browser redirects the user to an SSO URL
+
+// 4. Users sends request and gets a SAML resposne
+
+// 5. Auth0 returns the encoded SAML response to the browser
+
+// 6. The browser sends the SAML response to service provider for verification
+/*
+function verifySAMLRequest() {
```

Comment:
Remove this function comment (I will take up this stuff in the README or Technical analyses)